src/immutable/Stack.mo

```diff
@@ -106,15 +141,13 @@ module {
     todo()
   };
 
-  public func generate<T>(n : Nat, f : Nat -> T) : Stack<T> {
-    todo()
-  };
+  public func generate<T>(n : Nat, f : Nat -> T) : Stack<T> =
+    if (n == 0) null else ?(generate(/*(with underflow = false)*/ n - 1, f), f n);
 
   public func singleton<T>(item : T) : Stack<T> = ?(null, item);
 
-  public func repeat<T>(item : T, n : Nat) : Stack<T> {
-    todo()
-  };
+  public func repeat<T>(item : T, n : Nat) : Stack<T> =
+    if (n == 0) null else ?(repeat(item, /*(with underflow = false)*/ n - 1), item);
```

Comment:
What is `/*(with underflow = false)*/`? Is this future syntax that would deactivate underflow checking?